### DISCUSSION TITLE
Words separated into letters when they shouldn't have

### DISCUSSION THREAD

**Discussion by scythe837** [jsvine/pdfplumber#1113] [Ask for help with specific PDFs]
I have some odd text files which on some points have the words seen as separated letters no matter how much x_tolerance I would add even with huge values (as seen from the added image). 
Any idea what is the issue? or what to try else? Some words work ok by default, but some just stay as separate letters. 

![P390245_page_1](https://github.com/jsvine/pdfplumber/assets/71078144/d153428f-0b33-496b-9abd-269ebcfd7402) 
![image](https://github.com/jsvine/pdfplumber/assets/71078144/e856897d-6b27-4082-93eb-0d94ad8c57ea)

**Comment by jsvine** [comment]
Without the PDF, it's difficult to answer this definitively, but my initial guess is that the PDF itself has a bunch of space characters jammed between those letters. You can see a similar example and solution here: https://github.com/jsvine/pdfplumber/discussions/1099#discussioncomment-8751877

**Reply by scythe837** [reply]
[P390245.pdf](https://github.com/jsvine/pdfplumber/files/14590923/P390245.pdf)

frankly it looks that there is no space over there ... GEOCOMPOZIT should be one word and by the looks of the x0 and x1 they are ok with no spaces...

> {'text': 'FREZARE', 'x0': 28.346, 'x1': 61.946, 'top': 249.70800000000003, 'doctop': 249.70800000000003, 'bottom': 257.708, 'upright': True, 'direction': 1}
{'text': 'MIXTURA', 'x0': 66.746, 'x1': 100.34599999999999, 'top': 249.70800000000003, 'doctop': 249.70800000000003, 'bottom': 257.708, 'upright': True, 'direction': 1}
{'text': 'ASFALTICA', 'x0': 105.14599999999999, 'x1': 148.34599999999998, 'top': 249.70800000000003, 'doctop': 249.70800000000003, 'bottom': 257.708, 'upright': True, 'direction': 1}
{'text': '4', 'x0': 153.14599999999996, 'x1': 157.94599999999997, 'top': 249.70800000000003, 'doctop': 249.70800000000003, 'bottom': 257.708, 'upright': True, 'direction': 1}
{'text': 'CM', 'x0': 162.74599999999998, 'x1': 172.346, 'top': 249.70800000000003, 'doctop': 249.70800000000003, 'bottom': 257.708, 'upright': True, 'direction': 1}
{'text': '2', 'x0': 28.346, 'x1': 34.346000000000004, 'top': 259.704, 'doctop': 259.704, 'bottom': 269.704, 'upright': True, 'direction': 1}
{'text': '2', 'x0': 68.096, 'x1': 74.096, 'top': 259.704, 'doctop': 259.704, 'bottom': 269.704, 'upright': True, 'direction': 1}
{'text': '3', 'x0': 74.096, 'x1': 80.096, 'top': 259.704, 'doctop': 259.704, 'bottom': 269.704, 'upright': True, 'direction': 1}
{'text': 'M', 'x0': 166.346, 'x1': 172.346, 'top': 259.704, 'doctop': 259.704, 'bottom': 269.704, 'upright': True, 'direction': 1}
{'text': 'P', 'x0': 172.346, 'x1': 178.346, 'top': 259.704, 'doctop': 259.704, 'bottom': 269.704, 'upright': True, 'direction': 1}
{'text': '1', 'x0': 257.536, 'x1': 263.096, 'top': 259.714, 'doctop': 259.714, 'bottom': 269.714, 'upright': True, 'direction': 1}
{'text': '2', 'x0': 263.096, 'x1': 268.656, 'top': 259.714, 'doctop': 259.714, 'bottom': 269.714, 'upright': True, 'direction': 1}
{'text': '60,00000', 'x0': 268.656, 'x1': 310.346, 'top': 259.714, 'doctop': 259.714, 'bottom': 269.714, 'upright': True, 'direction': 1}
{'text': 'G', 'x0': 28.346, 'x1': 33.146, 'top': 273.708, 'doctop': 273.708, 'bottom': 281.708, 'upright': True, 'direction': 1}
{'text': 'E', 'x0': 33.146, 'x1': 37.946, 'top': 273.708, 'doctop': 273.708, 'bottom': 281.708, 'upright': True, 'direction': 1}
{'text': 'O', 'x0': 37.946, 'x1': 42.745999999999995, 'top': 273.708, 'doctop': 273.708, 'bottom': 281.708, 'upright': True, 'direction': 1}
{'text': 'C', 'x0': 42.745999999999995, 'x1': 47.54599999999999, 'top': 273.708, 'doctop': 273.708, 'bottom': 281.708, 'upright': True, 'direction': 1}
{'text': 'O', 'x0': 47.546, 'x1': 52.346, 'top': 273.708, 'doctop': 273.708, 'bottom': 281.708, 'upright': True, 'direction': 1}
{'text': 'M', 'x0': 52.346000000000004, 'x1': 57.146, 'top': 273.708, 'doctop': 273.708, 'bottom': 281.708, 'upright': True, 'direction': 1}
{'text': 'P', 'x0': 57.146, 'x1': 61.946, 'top': 273.708, 'doctop': 273.708, 'bottom': 281.708, 'upright': True, 'direction': 1}
{'text': 'O', 'x0': 61.946, 'x1': 66.746, 'top': 273.708, 'doctop': 273.708, 'bottom': 281.708, 'upright': True, 'direction': 1}
{'text': 'Z', 'x0': 66.746, 'x1': 71.54599999999999, 'top': 273.708, 'doctop': 273.708, 'bottom': 281.708, 'upright': True, 'direction': 1}

**Reply by jsvine** [reply]
If you use the code in the linked thread, I believe you'll find that there are indeed many extra space characters in the text, and that the code suggested there should fix it for you. Let me know if you run into trouble with that, though.

**Reply by scythe837** [reply]
Thank you, i suppose that was it. It solved it for this particular pdf.
